HPXMLtoOpenStudio/resources/airflow.rb

```diff
@@ -1399,77 +1425,241 @@ def self.add_ee_for_vent_fan_power(model, obj_name, frac_lost, is_cfis, pow = 0.
-    infil_program.addLine('Set ERVCpMin = (@Min ERVSupCp ERVSecCp)')
-    infil_program.addLine("Set ERVSupOutTemp = ERVSupInTemp + ERVCpMin/ERVSupCp * #{vent_mech_sens_eff} * (ERVSecInTemp - ERVSupInTemp)")
-    infil_program.addLine("Set ERVSupOutW = ERVSupInW + ERVCpMin/ERVSupCp * #{vent_mech_lat_eff} * (ERVSecInW - ERVSupInW)")
-    infil_program.addLine('Set ERVSupOutEnth = (@HFnTdbW ERVSupOutTemp ERVSupOutW)')
-    infil_program.addLine('Set ERVSensHeatTrans = ERV_MFR * ERVSupCp * (ERVSupOutTemp - ERVSupInTemp)')
-    infil_program.addLine('Set ERVTotalHeatTrans = ERV_MFR * (ERVSupOutEnth - ERVSupInEnth)')
-    infil_program.addLine('Set ERVLatHeatTrans = ERVTotalHeatTrans - ERVSensHeatTrans')
-
-    # Load calculation
-    infil_program.addLine('Set ERVTotalToLv = ERV_MFR * (ERVSupOutEnth - ERVSecInEnth)')
-    infil_program.addLine('Set ERVSensToLv = ERV_MFR * ERVSecCp * (ERVSupOutTemp - ERVSecInTemp)')
-    infil_program.addLine('Set ERVLatToLv = ERVTotalToLv - ERVSensToLv')
+    infil_program.addLine("Set Q_oa = #{oa_flow_rate_var_name}")
+    infil_program.addLine("Set Q_tot = #{tot_flow_rate_var_name}")
+    infil_program.addLine('Set Fan_MFR = Q_oa * OASupRho')
+
+    if (vent_mech_sens_eff > 0.0) || (vent_mech_lat_eff > 0.0)
+      # ERV Heat exchanger calculation
+      infil_program.addLine('Set ERVCpMin = (@Min OASupCp ZoneCp)')
+      infil_program.addLine("Set ERVSupOutTemp = OASupInTemp + ERVCpMin/OASupCp * #{vent_mech_sens_eff} * (ZoneTemp - OASupInTemp)")
+      infil_program.addLine("Set ERVSupOutW = OASupInW + ERVCpMin/OASupCp * #{vent_mech_lat_eff} * (ZoneW - OASupInW)")
+      infil_program.addLine('Set ERVSupOutEnth = (@HFnTdbW ERVSupOutTemp ERVSupOutW)')
+      infil_program.addLine('Set ERVSensHeatTrans = Fan_MFR * OASupCp * (ERVSupOutTemp - OASupInTemp)')
+      infil_program.addLine('Set ERVTotalHeatTrans = Fan_MFR * (ERVSupOutEnth - OASupInEnth)')
+      infil_program.addLine('Set ERVLatHeatTrans = ERVTotalHeatTrans - ERVSensHeatTrans')
+
+      # Load calculation
+      infil_program.addLine('Set FanTotalToLv = Fan_MFR * (ERVSupOutEnth - ZoneAirEnth)')
+      infil_program.addLine('Set FanSensToLv = Fan_MFR * ZoneCp * (ERVSupOutTemp - ZoneTemp)')
+      infil_program.addLine('Set FanLatToLv = FanTotalToLv - FanSensToLv')
+    else
+      # Load calculation
+      infil_program.addLine('Set FanTotalToLv = Fan_MFR * (OASupInEnth - ZoneAirEnth)')
+      infil_program.addLine('Set FanSensToLv = Fan_MFR * ZoneCp * (OASupInTemp - ZoneTemp)')
+      infil_program.addLine('Set FanLatToLv = FanTotalToLv - FanSensToLv')
+    end
 
     # Actuator
-    infil_program.addLine("Set #{erv_sens_load_actuator.name} = #{erv_sens_load_actuator.name} + ERVSensToLv")
-    infil_program.addLine("Set #{erv_lat_load_actuator.name} = #{erv_lat_load_actuator.name} + ERVLatToLv")
-    infil_program.addLine("Set QWHV_ervhrv = QWHV_ervhrv + #{UnitConversions.convert(vent_mech_cfm, 'cfm', 'm^3/s')}")
+    if not preconditioned
```

Comment:
This condition is applied only if ERV/HRV is called out of the preconditioning program/subprogram. If the ERV/HRV is a shared preconditioned system, this method will be called in preconditioning method with preconditioned = true, shouldn't apply the actuator and QWHV in that case because preconditioning program will handle that later.